modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalSpec.java

```diff
@@ -42,11 +51,14 @@
  * */
 
 public class RankEvalSpec extends ToXContentToBytes implements Writeable {
+    private static final Logger logger = Loggers.getLogger(RankEvalSpec.class);
 
     /** Collection of query specifications, that is e.g. search request templates to use for query translation. */
     private Collection<RatedRequest> ratedRequests = new ArrayList<>();
     /** Definition of the quality metric, e.g. precision at N */
     private RankedListQualityMetric metric;
+    /** optional: Template to base test requests on */
+    private Script script;
```

Comment:
For lack of a better line to comment this: I think script is missing from hashCode/equals